server/src/test/java/io/crate/execution/dml/IndexerTest.java

```diff
@@ -945,6 +944,32 @@ z timestamp default now()
         assertThat((long) insertValues[1]).isGreaterThanOrEqualTo(now);
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    public void test_adds_non_deterministic_sub_columns_when_top_is_not_in_targets() throws Exception {
```

Comment:
Maybe better: ```suggestion
    public void test_adds_non_deterministic_sub_columns_when_root_is_not_in_targets() throws Exception {
```

or `parent`?

Reply:
renamed to `test_adds_non_deterministic_sub_columns_when_root_is_not_in_targets`